pyomo/solvers/plugins/solvers/scip_direct.py

```diff
@@ -0,0 +1,838 @@
+            )
+        else:
+            scip_cons = self._solver_model.addConsSOS2(
+                scip_vars, weights=weights, name=conname
+            )
+        self._pyomo_con_to_solver_con_expr_map[con] = scip_cons
+        self._pyomo_con_to_solver_con_map[con] = scip_cons.name
+        self._solver_con_to_pyomo_con_map[conname] = con
+
+    def _scip_vtype_from_var(self, var):
+        """
+        This function takes a pyomo variable and returns the appropriate SCIP variable type
+        :param var: pyomo.core.base.var.Var
+        :return: B, I, or C
+        """
+        if var.is_binary():
+            vtype = "B"
+        elif var.is_integer():
+            vtype = "I"
+        elif var.is_continuous():
+            vtype = "C"
+        else:
+            raise ValueError(
+                "Variable domain type is not recognized for {0}".format(var.domain)
+            )
+        return vtype
+
+    def _set_objective(self, obj):
+        if self._objective is not None:
+            for var in self._vars_referenced_by_obj:
+                self._referenced_variables[var] -= 1
+            self._vars_referenced_by_obj = ComponentSet()
+            self._objective = None
+
+        if obj.active is False:
+            raise ValueError("Cannot add inactive objective to solver.")
+
+        if obj.sense == minimize:
+            sense = "minimize"
+        elif obj.sense == maximize:
+            sense = "maximize"
+        else:
+            raise ValueError("Objective sense is not recognized: {0}".format(obj.sense))
+
+        scip_expr, referenced_vars = self._get_expr_from_pyomo_expr(
+            obj.expr, self._max_obj_degree
+        )
+
+        for var in referenced_vars:
+            self._referenced_variables[var] += 1
+
+        self._solver_model.setObjective(scip_expr, sense=sense)
+        self._objective = obj
+        self._vars_referenced_by_obj = referenced_vars
+
+        self._needs_updated = True
+
+    def _postsolve(self):
+        # the only suffixes that we extract from SCIP are
+        # constraint duals, constraint slacks, and variable
+        # reduced-costs. scan through the solver suffix list
+        # and throw an exception if the user has specified
+        # any others.
+        extract_duals = False
+        extract_slacks = False
+        extract_reduced_costs = False
+        for suffix in self._suffixes:
+            flag = False
+            if re.match(suffix, "dual"):
+                extract_duals = True
+                flag = True
+            if re.match(suffix, "slack"):
+                extract_slacks = True
+                flag = True
+            if re.match(suffix, "rc"):
+                extract_reduced_costs = True
+                flag = True
+            if not flag:
+                raise RuntimeError(
+                    "***The scip_direct solver plugin cannot extract solution suffix="
+                    + suffix
+                )
```

Comment:
Design question: is it better to raise an error or to log (or print loudly) that other requests are being ignored?

(Note: I don't have a strong preference either way.)

Reply:
I would lean slightly towards the print loudly option, but can see the potential danger for users. As a larger preference though, I would like to stick to similar behavior with the other interfaces.